### DISCUSSION TITLE
how to add UserSelect into button

### DISCUSSION THREAD

**Discussion by whyrmph** [Rapptz/discord.py#9284] [Q&A]
Hi, I need the dropdown menu of **_discord.ui.UserSelect_** to be shown when the button is clicked
And so that after selecting the user, he was given access to the voice channel

I tried to do something, but nothing really works
Please help to implement

I really ask you to give an example of the code that implements this all please

In this case it is too difficult for me to deal with the documentation because there is not enough information

Thanks for understanding!

Code: 
 ```
class ButtonsVoiceManager(discord.ui.View):  
          
           def __init__(self, author: typing.Union[discord.Member, discord.User]):  
            self.author = author
            super().__init__(timeout=None)

           class MySelect(discord.ui.UserSelect):
             async def callback(self, interaction):
              self.xs = [i.id for i in self.values]
              self.y = ''.join(str(x) for x in self.xs)   
       @discord.ui.button(label="access",style=discord.ButtonStyle.gray,emoji="🪙",custom_id='persistent_view:access')
           async def access_button(self,interaction:discord.Interaction,button:discord.ui.Button):
            conn = sqlite3.connect('voice.db')
            c = conn.cursor()
            id = interaction.user.id
            c.execute("SELECT voiceID FROM voiceChannel WHERE userID = ?", (id,))
            voice=c.fetchone()
            if voice is None:
             embed = discord.Embed(title="А ну стой!", description="",color=0x7289da)
             embed.set_author(name=f"{interaction.guild.me.display_name}", icon_url=f"{interaction.guild.me.display_avatar.url}")     
             embed.add_field(name=f'', value=f'{interaction.user.mention}, настраивать комнату можно только находясь в ней!')  
             embed.set_image(url="https://media.discordapp.net/attachments/1037473534257143891/1069053206606397480/deny.png")              
             await interaction.response.send_message(embed=embed, ephemeral=True)
            else:
             await interaction.response.send_message('test', view=discord.ui.View().add_item(self.MySelect()))
             channelID = voice[0]
             channel = bot.get_channel(channelID)
             await channel.set_permissions(self.MySelect().y, connect=True)
             embed = discord.Embed(title="Всё получилось!", description="",color=0x7289da)
             embed.set_author(name=f"{interaction.guild.me.display_name}", icon_url=f"{interaction.guild.me.display_avatar.url}")     
             embed.add_field(name=f'', value=f'{interaction.user.mention}, по Твоему запросу я выдала {self.MySelect().y} доступ в Твой милый голосовой канал! 🔒')  
             embed.set_image(url="https://media.discordapp.net/attachments/1037473534257143891/1069052970404155433/agree_v2.png")
             await interaction.response.send_message(embed=embed, ephemeral=True)
            conn.commit()
            conn.close()
```

**Comment by felipeyx** [comment]
The main issue is that every time you call:

self.MySelect()

you create a new select instance, so it does not contain the user selected in the previous instance.

Let the select callback apply the permission directly:

class UserAccessSelect(discord.ui.UserSelect):
    def __init__(self, channel: discord.VoiceChannel):
        super().__init__(
            placeholder="Select a user",
            min_values=1,
            max_values=1,
        )
        self.channel = channel

    async def callback(self, interaction: discord.Interaction):
        member = self.values[0]

        await self.channel.set_permissions(
            member,
            connect=True,
        )

        await interaction.response.edit_message(
            content=f"{member.mention} can now connect.",
            view=None,
        )


class UserAccessView(discord.ui.View):
    def __init__(self, channel):
        super().__init__(timeout=60)
        self.add_item(UserAccessSelect(channel))

Then from the button callback:

channel = interaction.guild.get_channel(channelID)

await interaction.response.send_message(
    "Choose a user:",
    view=UserAccessView(channel),
    ephemeral=True,
)

This way the selected Member is available directly inside the select callback and you don't need to store IDs on separate temporary UserSelect instances.
